blueprints/enhanced-physlocations.md

```diff
@@ -0,0 +1,352 @@
+
+### Traffic Ops Impact
+<!--
+*How* will this impact Traffic Ops (at a high level)?
+-->
+
+The primarily changes take place within Traffic ops. There will be an API response change for the physical location endpoints, a database table addition and modification.
+
+#### REST API Impact
+<!--
+*How* will this impact the Traffic Ops REST API?
+
+What new endpoints will be required?
+How will existing endpoints be changed?
+What will the requests and responses look like?
+What fields are required or optional?
+What are the defaults for optional fields?
+What are the validation constraints?
+-->
+
+This modification will update the `GET` response and `POST` request fields for the `/phys_locations` endpoint. It should be a breaking change modifying the names of existing fields and adding new fields.
+
+The `GET` response will be modified as follows.
+
+```
+{ "response": [
+    {
+        "comments": "",
+        "id": 2,
+        "lastUpdated": "2018-12-05 17:50:58+00",
+        "name": "CDN_in_a_Box",
+        "regionId": 1,
+        "region": "Washington, D.C",
+        "shortName": "ciab",
+        "addresses": [ {
+        		"address1": "1600 Pennsylvania Avenue NW",
+		 		"address2": "",
+		 		"address3": "",
+		 		"address4": "",
+        		"locality": "Washington",
+        		"administrativeArea": "DC",
+        		"postalCode": "20500",
+        		"longitude": -77.03650834403494,
+        		"latitude": 38.897846677690815,
+        		"type": "primary"
+        }],
+        "contacts": [ {
+        		"contact": "Joe Schmo",
+        		"method": "phone",
+        		"value": "800-555-1234",
+        		"comments": "Security Guard",
+			"priority": 1,
+			"purpose": "controls access to physical location"
+        }]
+    }
+]}
+```
+
+The `POST` request will look very similar. 
+
+```
+{
+    "comments": "",
+    "name": "CDN_in_a_Box",
+    "regionId": 1,
+    "region": "Washington, D.C",
+    "shortName": "ciab",
+    "addresses": [ {
+    		"address1": "1600 Pennsylvania Avenue NW",
+	 		"address2": "",
+	 		"address3": "",
+	 		"address4": "",
+    		"locality": "Washington",
+    		"administrativeArea": "DC",
+    		"postalCode": "20500",
+    		"longitude": -77.03650834403494,
+    		"latitude": 38.897846677690815,
+    		"type": "primary"
+    }],
+    "contacts": [ {
+    		"contact": "Joe Schmo",
+     		"method": "phone",
+    		"value": "800-555-1234",
+    		"comments": "Security Guard",
+		"priority": 1,
+		"purpose": "controls access to physical location"
+    }]
+}
+
+```
+
+For the POST request body it will be important to validate the longitude and latitude as sensible values.
+
+Some of the address lines can be optional.
+
+The type and method fields could probably be enumeration types. But I am not sure we understand/want to enforce all of the possible use cases so it may be important to limit the input text length. 
```

Comment:
I think we should address whether we'll be using an enumeration or not now, since it will affect behavior, i.e. whether or not new types/methods can be added at runtime.

Personally I like enums.

Reply:
Do you think we can understand enough of the use cases to make these enums?

I think for location `type` we could have an enum. However `method` for contact I think there are just to many communications methods to even attempt to enumerate them unless we want to be unhelpfully generic.